pkg/upgrade/upgrade.go

```diff
@@ -40,9 +42,19 @@ import (
 	"github.com/opendatahub-io/opendatahub-operator/v2/components/workbenches"
 	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/cluster"
 	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/deploy"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/gvk"
 	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/metadata/labels"
 )
 
+type JobSpec struct {
+	Gvk       schema.GroupVersionKind
+	Namespace string
+	// path to the field, like "metadata", "name"
+	Path []string
+	// set of values for the field to match object, any one matches
+	Values []string
+}
```

Comment:
Is it a `JobSpec` or `ResourceSpec`? It does not carry any job-specific traits (yet). Perhaps in #957 we could split `ResourceSpec` to carry what you defined here + matcher and expand on the `JobSpec` with retry and actual func doing the work. WDYT?